vector-im/element-android
# Save m.local_notification_settings.<device-id> event in account_data (PSG-873)

## Type of change


## Content


When background sync is used, on session start and each time we switch on background sync, account data event "org.matrix.msc3890.local_notification_settings.<device-id>" is updated.
When switching from background sync to regular push notifications method, any existing account data event is updated with empty content (API to delete an account data event is not available yet).

Also in this PR: renaming some notifications related use cases for consistency.

## Motivation and context


Closes #7596 

## Screenshots / GIFs





## Tests



Tests should be performed with both the GPlay and FDroid variant.
- Install Ntfy app
- Open the Network plugin in Flipper
- Check the network call is correct after signin
- Check the network call is correct when toggling the notifications for the current session in the settings
- (Only FDroid) Check the network call is correct when switching from background sync mode to push notification method changing the related setting (Notifications -> Notifications method)

## Tested devices

- OS version(s): Android 11

## Checklist
